source/elements/oneDAL/source/data_management/data_sources.rst

```diff
@@ -1,7 +1,117 @@
+.. default-domain:: cpp
+
 .. _data-sources:
 
 ============
 Data Sources
 ============
 
-TBD
+This section describes the types related to the :txtref:`data-source` concept.
+
+---------------
+CSV data source
+---------------
+Class ``csv::data_source`` is an api for accessing the data source represented as a :capterm:`csv file <CSV file>`.
+CSV data source can be used in :expr:`read` operation to extract data in text format from the given input file,
+process it using provided parameters (such as delimiter and parsing options), transform it into numerical representation,
+and store it as an in-memory :txtref:`dataset` of a chosen type.
+
+Supported type of in-memory dataset for :expr:`read` operation with CSV data source is :txtref:`table`.
+
+CSV data source requires input file name to be set in the constructor, while other parameters can be initialized with default values.
+
+-------------
+Usage example
+-------------
+::
+
+   const auto data_source = onedal::csv::data_source("data.csv", ',');
+
+   const auto table = ondedal::read<onedal::table>(data_source);
+
+
+---------------------
+Programming Interface
+---------------------
+All type and function in this section shall be declared in the
+``oneapi::dal::csv`` namespace and be available via inclusion of the
+``oneapi/dal/io/csv.hpp`` header file.
+
+::
+
+   enum class read_options {
+      default = 0,
+      parse_header = 1 << 0
+   };
+
+   class data_source {
+   public:
+      static constexpr char default_delimiter = ',';
+      static constexpr read_options default_read_options = read_options::default;
+
+      data_source(const char *file_name,
+                  char delimiter = default_delimiter,
+                  read_options opts = default_read_options);
+
+      data_source(const std::string &file_name,
+                  char delimiter = default_delimiter,
+                  read_options opts = default_read_options);
+
+      std::string get_file_name() const;
+      char get_delimiter() const;
+      read_options get_read_options() const;
+   };
+
+.. namespace:: onedal::csv
+.. class:: data_source
+
+   .. function:: data_source(const char *file_name, char delimiter = default_delimiter, read_options opts = default_read_options)
+
+      Creates new instance of a csv data source with the given :expr:`file_name`, :expr:`delimiter` and read options :expr:`opts` flag.
+
+   .. function:: data_source(const std::string &file_name, char delimiter = default_delimiter, read_options opts = default_read_options);
+
+      Creates new instance of a csv data source with the given :expr:`file_name`, :expr:`delimiter` and read options :expr:`opts` flag.
+
+   .. member:: std::string file_name = ""
+
+      String which contains name of the file with the dataset for read.
+
+      Getter
+         | ``std::string get_filename() const``
+
+   .. member:: char delimiter = ','
+
+      Symbol that represents delimiter between continuous feature columns in the input file.
```

Comment:
that separate features in the input file?

Reply:
Done.